File: CovidTracerApp/app/src/main/java/com/example/covidtracerapp/database/ContactedDatabase.java
package com.example.covidtracerapp.database;

import android.content.Context;
import android.util.Log;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

@Database(entities = {ContactedEntity.class}, version = 1, exportSchema = false)
@TypeConverters(DateConverter.class)
public abstract class ContactedDatabase extends RoomDatabase {

    public abstract ContactedDAO contactedDAO();

    private static final String DB_NAME = "contacted.db";
    private static final String LOG_TAG = ContactedDatabase.class.getSimpleName();
    private static ContactedDatabase sInstance;

    public static ContactedDatabase getsInstance(Context context) {
        if (sInstance == null){
            synchronized (ContactedDatabase.class){
                if (sInstance == null) {
                    Log.d(LOG_TAG, "Creating a new database instance");
                    sInstance = Room.databaseBuilder(context.getApplicationContext(), ContactedDatabase.class, DB_NAME)
                            .build();
                }
            }
        }
        Log.d(LOG_TAG, "Getting the database instance");
        return sInstance;
    }
}

File: CovidTracerApp/app/src/main/java/com/example/covidtracerapp/database/ContactedDatabase.kt
package com.example.covidtracerapp.database

import android.content.Context
import android.util.Log
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

@Database(entities = [ContactedEntity::class], version = 1, exportSchema = false)
@TypeConverters(
    DateConverter::class
)
abstract class ContactedDatabase : RoomDatabase() {
    abstract fun contactedDAO(): ContactedDAO?

    companion object {
        private const val DB_NAME = "contacted.db"
        private val LOG_TAG = ContactedDatabase::class.java.simpleName
        private var sInstance: ContactedDatabase? = null
        fun getsInstance(context: Context): ContactedDatabase? {
            if (sInstance == null) {
                synchronized(ContactedDatabase::class.java) {
                    if (sInstance == null) {
                        Log.d(
                            LOG_TAG,
                            "Creating a new database instance"
                        )
                        sInstance =
                            Room.databaseBuilder(
                                context.applicationContext,
                                ContactedDatabase::class.java,
                                DB_NAME
                            )
                                .build()
                    }
                }
            }
            Log.d(LOG_TAG, "Getting the database instance")
            return sInstance
        }
    }
}